tests/test_whole_archive/testlib.c

```diff
@@ -0,0 +1,9 @@
+// Library containing only single static constructor function.
+// This is used to verify that the -Wl,-whole-archive works as expected
+
+extern int foo;
+
+void library_ctor() __attribute__ ((constructor));
```

Comment:
Any reason the declaration is separate from the definition here?

Reply:
I believe that __attribute__ is required to be on the declaration.